R/misc_yhat.R

```diff
@@ -43,6 +43,8 @@ yhat.lm <- function(X.model, newdata, ...) {
 yhat.randomForest <- function(X.model, newdata, ...) {
   if (X.model$type == "classification") {
     pred <- predict(X.model, newdata, type = "prob", ...)
+    # if result is a vector then ncol parameter is null
+    if (is.null(ncol(pred))) return(pred)
```

Comment:
if there is a return() call in line 47, then also line 49 shall finish with return()

Reply:
it's not really necessary for line 49 to ends with a return. Line 47 has to because otherwise line 48 (check in if statement) would raise an error

Reply:
three if's with different flow/return paths - it is an inconsistent flow without a good reason.
it will be much more readable if you will call return() ass soon as the result is ready in each path